src/file_util.rs

```diff
@@ -33,29 +33,82 @@ pub fn read_visible_entries(dir: &Path) -> Result<Vec<PathBuf>, Error> {
     Ok(res)
 }
 
-pub fn create_dir<P:AsRef<Path>>(path: P, recursive: bool) -> Result<(), Error>
+pub struct Dir<P: AsRef<Path>> {
+    path: P,
+    recursive: bool,
+    mode: Option<u32>,
+    uid: Option<uid_t>,
+    gid: Option<gid_t>,
+}
+
+impl<P> Dir<P> where P: AsRef<Path> {
+    pub fn new(path: P) -> Dir<P> {
+        Dir {
+            path: path,
+            recursive: false,
+            mode: None,
+            uid: None,
+            gid: None,
+        }
+    }
+
+    pub fn recursive(&mut self, recursive: bool) -> &mut Dir<P> {
+        self.recursive = recursive;
+        self
+    }
+
+    pub fn mode(&mut self, mode: u32) -> &mut Dir<P> {
+        self.mode = Some(mode);
+        self
+    }
+
+    pub fn uid(&mut self, uid: uid_t) -> &mut Dir<P> {
+        self.uid = Some(uid);
+        self
+    }
+
+    pub fn gid(&mut self, gid: gid_t) -> &mut Dir<P> {
+        self.gid = Some(gid);
+        self
+    }
+
+    pub fn create(&self) -> Result<(), Error> {
+        create_dir(self.path.as_ref(), self.recursive,
+            self.mode, self.uid, self.gid)
+    }
+}
+
+pub fn create_dir(path: &Path, recursive: bool,
```

Comment:
If you don't keep this function for backward compatibility (i.e. you change arguments anyway). then don't make it public anymore. If it's not public just make it a method and don't "unpack" structure items manually.